backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java

```diff
@@ -1371,6 +1383,66 @@ public final boolean isTaxIncluded(@NonNull final org.compiere.model.I_C_Invoice
 		return invoice.isTaxIncluded(); // 08486: use the invoice's flag, not whatever the PL sais right now
 	}
 
+	@Override
+	public void setInvoiceLineTaxes(final Properties ctx, @NonNull final I_C_Invoice invoice)
```

Comment:
the `fromCountryId` in this method is computed in a different way compared with MInvoiceLine.setTax() method.
Pls make sure we are always computing things in one way.

More, I would advise to introduce some getFromCountryId(I_C_Invoice invoice, I_C_InvoiceLine invoiceLine) method in IInvoiceBL.
You could call that method from both places.